core/server/worker/src/main/java/alluxio/worker/block/BlockMasterSync.java

```diff
@@ -214,55 +219,38 @@ private void handleMasterCommand(Command cmd) throws IOException, ConnectionFail
   }
 
   /**
-   * Thread to remove block from master.
+   * Thread to remove blocks that have been freed by the master.
    */
   @NotThreadSafe
   private class BlockRemover implements Runnable {
-    private final BlockWorker mBlockWorker;
-    private final long mSessionId;
-    private final long mBlockId;
-    private final Map<Long, Boolean> mRemovingBlockIdToFinished;
+    private final List<Long> mBlocks;
 
     /**
-     * Creates a new instance of {@link BlockRemover}.
-     *
-     * @param blockWorker block worker for data manager
-     * @param removingBlockIdToFinished map from block ID to whether it has been removed
-     * @param sessionId the session id
-     * @param blockId the block id
+     * @param blocks the blocks to remove
      */
-    public BlockRemover(BlockWorker blockWorker, Map<Long, Boolean> removingBlockIdToFinished,
-        long sessionId, long blockId) {
-      mBlockWorker = blockWorker;
-      mRemovingBlockIdToFinished = removingBlockIdToFinished;
-      mSessionId = sessionId;
-      mBlockId = blockId;
+    public BlockRemover(List<Long> blocks) {
+      mBlocks = blocks;
     }
 
     @Override
     public void run() {
-      boolean success = false;
-      try {
-        mBlockWorker.removeBlock(mSessionId, mBlockId);
-        success = true;
-        synchronized (mRemovingBlockIdToFinished) {
-          mRemovingBlockIdToFinished.put(mBlockId, true);
-        }
-        LOG.info("Block {} removed at session {}", mBlockId, mSessionId);
```

Comment:
Can you keep this logging, probably at debug level?

Reply:
done, thanks